### PR TITLE
Hotfix/7/SC-5287

### PR DESCRIPTION
**Pull request by dominikglandorf**
# Description
Due to changes in login logic, OAuth2 login challenges were not considered when redirecting after login.

## Links to Tickets or other pull requests
-

## Changes
Check for login_challenge in session when redirecting.

## Data Security <sub><sup>details [on Confluence](https://docs.schul-cloud.org/x/2S3GBg)</sup></sub>
Nothing new

## Deployment
Nothing new

## New Repos, NPM packages or vendor scripts
Nothing new

## Screenshots of UI changes
No change

## Approval for review
- [ x ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: controllers/login.js [hunk 1]
```diff
@@ -47,7 +47,6 @@ router.post('/login/', (req, res, next) => {
 		schoolId,
 		redirect,
 	} = req.body;
-	const validRedirect = redirectHelper.getValidRedirect(redirect);
```
Comment (janrenz) [inline controllers/login.js:50]: is this removed on purpose? (i know its not because we just had a call, so this is just a reminder)

**Review by janrenz — CHANGES_REQUESTED** [review]

File: config/http-headers.js [hunk 2]
```diff
@@ -129,6 +129,7 @@ const config = {
 		'Access-Control-Allow-Credentials': 'true',
 		'Referrer-Policy': 'same-origin',
 		'Feature-Policy': "vibrate 'self'; speaker *; fullscreen *; sync-xhr *; notifications 'self'; push 'self'; geolocation 'self'; midi 'self'; microphone 'self'; camera 'self'; magnetometer 'self'; gyroscope 'self'; payment 'none';",
+		'X-Frame-Options': 'allow-from https://acc.bettermarks.com/',
```
Comment (janrenz) [inline config/http-headers.js:132]: Please check with DevOps if this should be set on the instances or here in code

File: config/http-headers.js [hunk 3]
```diff
@@ -61,7 +61,7 @@ const config = {
 				defaultSrc: 'https://www10-fms.hpi.uni-potsdam.de https://schul-cloud-hpi.s3.hidrive.strato.com',
 			},
 			'^/courses': {
-				defaultSrc: 'https://nexboard.nexenio.com https://www.geogebra.org https://lti.tools https://codeocean.openhpi.de https://acc.bettermarks.com https://moodle.schul-cloud.org',
+				defaultSrc: 'https://oauth.hotfix7.schul-cloud.dev https://nexboard.nexenio.com https://www.geogebra.org https://lti.tools https://codeocean.openhpi.de https://acc.bettermarks.com https://moodle.schul-cloud.org',
```
Comment (Parnswir) [inline config/http-headers.js:64]: I don't think it's a good idea to merge this as-is. Maybe this should read this from the configuration?

**Review by Parnswir — CHANGES_REQUESTED** [review]

**Comment by dominikglandorf** [comment]
This was caused by another error in the login logic.

**Pull request closed without merging**